src/ChainSafe.Gaming/Web3/Core/Web3.cs

```diff
@@ -12,8 +12,9 @@ namespace ChainSafe.Gaming.Web3
 {
     /// <summary>
     /// Facade for all Web3-related services.
+    /// Use this as an entry point to all SDK features.
     /// </summary>
-    public class Web3 : IAsyncDisposable
+    public class Web3 : IAsyncDisposable, IInitializableWeb3
```

Comment:
Is this abstraction (`IIinitializableWeb3`) necessary? Since we're always using the concrete class we make the same calls from just web3, If there needs to be an abstraction I believe `IWeb3` makes sense instead `IInitializableWeb3` since we only have one concrete web3 type or do we plan on having more than one?

Reply:
It's there just to hide the `InitializeAsync' method as it should only be called by the `Web3Builder`. It also has `internal` access modifier.

Reply:
If that's the only reason maybe we should just change the accessor on the method, `internal InitializeAsync` should hide it on Unity.

Reply:
Extra base types can be confusing if there's only one inheriting type, if we plan on adding more Web3 Types like we have signer and txn executor types this is fine but if there'll only be just one concrete type it's a bit confusing, especially with an -able suffix like _initializable_ which makes me think is there a non-initializable type 😅

Reply:
> If that's the only reason maybe we should just change the accessor on the method, `internal InitializeAsync` should hide it on Unity.

You're right. Why didn't I think of that?

Reply:
Fixed

Reply:
Have you pushed the commit? I don't see any changes.

Reply:
Agree here on Initializable/Non-initializable points.

Reply:
Snap. My git software didn't push and didn't show me a msg box either because I was not logged in.
It's pushed now.

Reply:
For me personally, having a Zenject background, IInitializable means that the class has the `Initialize` method. So I don't agree with Initializable/Non-initializable points :D